operator/src/main/java/oracle/kubernetes/operator/DomainStatusUpdater.java

```diff
@@ -535,20 +535,10 @@ private boolean domainJustAvailable() {
       }
 
       private void setStatusConditions(DomainStatus status) {
-        if (allIntendedServersRunning()) {
-          status.removeConditionWithType(Failed);
-          status.addCondition(new DomainCondition(Completed).withStatus(TRUE));
-          status.addCondition(new DomainCondition(Available).withStatus(TRUE));
-        } else {
-          status.addCondition(new DomainCondition(Completed).withStatus(FALSE));
-          if (sufficientServersRunning()) {
-            status.addCondition(new DomainCondition(Available).withStatus(TRUE));
-          } else if (status.hasConditionWithType(Available)) {
-            status.addCondition(new DomainCondition(Available).withStatus(FALSE));
-          }
-
-          addTooManyReplicasFailures(status);
-        }
+        status.addCondition(new DomainCondition(Completed).withStatus(isProcessingCompleted() ? TRUE : FALSE));
```

Comment:
It looks like status for domain condition `Completed` is set to `TRUE`  when `isProcessingCompleted()` returns true. However, DOMAIN_COMPLETE event is generated when `processingJustCompleted()`  returns true. Should the DOMAIN_COMPLETE event be generated only when the status for domain condition `Completed` is set to `TRUE` ?

Reply:
`processingJustCompleted()` indicates a transition from Completed/false to Completed/true. The logic in this method computes the new overall conditions, which may be the same as the previous ones, in which case no such event is generated.